spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/expressions/TransformExpressions.scala

```diff
@@ -119,3 +118,38 @@ case class IcebergBucketTransform(numBuckets: Int, child: Expression) extends Ic
 
   override def dataType: DataType = IntegerType
 }
+
+case class IcebergTruncateTransform(child: Expression, width: Int) extends IcebergTransformExpression {
+
+  override def children: Seq[Expression] = child :: Nil
+
+  @transient lazy val truncateFunc: Any => Any = child.dataType match {
+    case _: DecimalType =>
+      val t = Transforms.truncate[Any](icebergInputType, width)
```

Comment:
Can we define this as follows?

```
val t = Transforms.truncate[java.math.BigDecimal](icebergInputType, length)
d: Any =>
  val truncatedValue = t(d.asInstanceOf[Decimal].toJavaBigDecimal)
  Decimal(truncatedValue)
```

I think we can get rid of the extra cast by using `Transforms.truncate[java.math.BigDecimal]`.